compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -3912,6 +3961,8 @@ object Types {
       tp.derivedSuperType(thistp, supertp)
     protected def derivedAppliedType(tp: AppliedType, tycon: Type, args: List[Type]): Type =
       tp.derivedAppliedType(tycon, args)
+    protected def derivedAppliedTerm(tp: AppliedTermRef, fn: Type, args: List[Type]): Type =
```

Comment:
should be `derivedAppliedTermRef` for consistency (or rename `AppliedTermRef` to `AppliedTerm` maybe)